src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Instance.hs

```diff
@@ -164,18 +171,33 @@ instance {-# OVERLAPPING #-} (SymbolicData a x, n ~ TypeSize a x, KnownNat n) =>
                 V.zipWithM (\x y -> newAssigned $ \p -> p bs * (p x - p y) + p y) ts fs
 
 -- TODO: make a proper implementation of Arbitrary
-instance (Arithmetic a, KnownNat n) => Arbitrary (ArithmeticCircuit n a) where
-    arbitrary = return $ c1 * c2
-        where
-            c1, c2 :: ArithmeticCircuit n a
-            c1 = ArithmeticCircuit { acCircuit = mempty, acOutput = pure 1}
-            c2 = ArithmeticCircuit { acCircuit = mempty, acOutput = pure 2}
-
+instance (Arithmetic a, Arbitrary a) => Arbitrary (ArithmeticCircuit 1 a) where
+    arbitrary = do
+        k <- integerToNatural <$> chooseInteger (0, 100)
+        let ac = ArithmeticCircuit { acCircuit = mempty {acInput = [1..k]}, acOutput = pure k }
+        arbitrary' ac 0
+
+arbitrary' :: forall a . (Arithmetic a, Arbitrary a, FromConstant a a) => ArithmeticCircuit 1 a -> Natural -> Gen (ArithmeticCircuit 1 a)
+arbitrary' ac 20 = return ac
+arbitrary' ac iter = do
+    let vars = getAllVars . acCircuit $ ac
+    li <- oneof $ fmap return vars
+    ri <- oneof $ fmap return vars
+    let (l, r) =( ac { acOutput = pure li }, ac { acOutput = pure ri })
+    ac' <- oneof $ fmap return [
+        l + r
+        , l * r
+        , l - r
+        , r - l
+        , l // r
+        , r // l
```

Comment:
Duplicating division and subtraction makes them twice as likely to be chosen. Since `l` and `r` are already random, maybe we want equal chances for each operation?